test/test.js

```diff
@@ -23,7 +23,7 @@ request.Test.prototype.expectJSON = function(json, done) {
 request.Test.prototype.expectNoHeader = function(header, done) {
   this.expect(function(res) {
     if (header.toLowerCase() in res.headers) {
-      return 'Unexpected header in response: ' + header;
+      return new Error('Unexpected header in response: ' + header);
```

Comment:
For future reference, this change fixes the issue because the supertest library treats `Error` instances as errors (whether returned or thrown), per https://github.com/visionmedia/supertest/blob/v2.0.0/lib/test.js#L265-L269.